x-pack/plugins/security_solution/public/common/store/store.ts

```diff
@@ -104,6 +107,15 @@ export const createStoreFactory = async (
       ...subPlugins.timelines.store.initialState.timeline!,
       timelineById: {
         ...subPlugins.timelines.store.initialState.timeline.timelineById,
+        ...addNewTimeline({
```

Comment:
If you go to OsQuery or ML and try to attach anything to cases a type error occurs when you navigate to the case by clicking the "View case" link in the toaster. This is happening only to Security solution. The reason is, that a timeline is not initialized early enough and the timeline is empty. Specifically in this line here https://github.com/elastic/kibana/blob/7745d367038b167783fc27b1df794dc5c696cc23/x-pack/plugins/security_solution/public/timelines/components/modal/header/index.tsx#L83 the selectors returns `undefined`. This is why we need to initialize the store with a default timeline.

Before it was not an issue, because cases were [waiting](https://github.com/elastic/kibana/pull/177798/files#diff-550be83d355c4b74e15efc1ac9396efac66c9820878b5b6d27a6e8de2fb737c4L151) for the `appId` to be available before rendering the `CasesContext` and all its children. In this PR, to stop the flakiness on our tests and optimize the cases app, the `CasesContext` and all its children [render](https://github.com/elastic/kibana/pull/177798/files#diff-550be83d355c4b74e15efc1ac9396efac66c9820878b5b6d27a6e8de2fb737c4R148) immediately. If you go to OsQuery or ML and try to attach anything to cases a type error occurs when you navigate to the case by clicking the "View case" link in the toaster. This is happening only to Security solution. The reason is, that a timeline is not initialized early enough and the timeline is empty. Specifically in this line here https://github.com/elastic/kibana/blob/7745d367038b167783fc27b1df794dc5c696cc23/x-pack/plugins/security_solution/public/timelines/components/modal/header/index.tsx#L83 the selectors returns `undefined`. This is why we need to initialize the store with a default timeline.

Before it was not an issue, because cases were [waiting](https://github.com/elastic/kibana/pull/177798/files#diff-550be83d355c4b74e15efc1ac9396efac66c9820878b5b6d27a6e8de2fb737c4L151) for the `appId` to be available before rendering the `CasesContext` and all its children. In this PR, to stop the flakiness on our tests and optimize the cases app, the `CasesContext` and all its children [render](https://github.com/elastic/kibana/pull/177798/files#diff-550be83d355c4b74e15efc1ac9396efac66c9820878b5b6d27a6e8de2fb737c4R148) immediately.

Reply:
This makes sense @cnasikas, `timelineById` creations are dispatched from the components in the rendering cycle, so we can not assume a timeline is always created, if we do so and access it before the creation action is dispatched the app crashes. 

@kqualters-elastic @michaelolo24 
The approach implemented here, having the timeline defaults created in the store initialization, makes sense to me, but if we follow this approach maybe we should do the same for all the timeline ids, including the `TimelineId.casePage` defined above, and the `test` timeline, right?

And if we do it this way, I think it would be better to define all the timelines in a static way in the `initialTimelineState` constant that now defaults to `EMPTY_TIMELINE_BY_ID = {}`:
https://github.com/elastic/kibana/blob/a79a4d861f40d904ae0bc84e6edf32270a33ae96/x-pack/plugins/security_solution/public/timelines/store/reducer.ts#L111-L115

what do you think?

Reply:
I mentioned to @cnasikas that imo we should just remove TimelineId.cases, this id was from when ids were tied to all of the tables in the app, and all tables were 'timelines'. The timeline in the bottom bar on the cases page is TimelineId.active, and tables have a separate id if needed. Only table on the cases page is the alerts table. TimelineId.test shouldn't ever be a thing outside of unit tests as I understand, but I guess it can't hurt? ha. And yes I agree on the main point, defining a bare minimum set of state statically so that the app doesn't crash we should absolutely do. Creating the store immediately with a minimal static initial state, and then populating things like sourcerer data and timelines if needed via actions is tech debt/feature/bug depending on who is asking and something I'd like to clean up soon.

Reply:
Okay, let's keep this as it is for now, and move/clean stuff when we refactor the timeline store.